src/frontend/src/lib/utils/token.utils.ts

```diff
@@ -141,6 +141,42 @@ export const mapTokenUi = ({
 	})
 });
 
+/** Function to sum the balances of two tokens.
+ *
+ * If the decimals of the tokens are the same, the balances are added together.
+ * If the decimals are different, the function returns null.
+ * If one of the balances is nullish, the function returns the other balance.
+ * If both balances are nullish, the function prioritize the first token
+ * NOTE: the function assumes that the two tokens are always 1:1 twins, for example BTC and ckBTC, or ETH and SepoliaETH
+ *
+ * @param token1
+ * @param token2
+ * @returns The sum of the balances or nullish value.
+ */
+export const sumTokenBalances = ([token1, token2]: TokenUi[]): TokenUi['balance'] =>
+	nonNullish(token1.balance) && nonNullish(token2.balance)
+		? token1.decimals === token2.decimals
+			? token1.balance.add(token2.balance)
+			: null
+		: isNullish(token2.balance)
```

Comment:
what if token2 balance is zero but token1 balance is >0 ?

Reply:
same question for usd

Reply:
it will be added in both cases, so it will stay the same as the other. I didn't put the check for a zero balance, do you think it is necessary? Meaning, either way, summing 0 is ok no?

Reply:
I oversaw it, good point.